client/src/components/DisplayTable/DisplayTable.tsx

```diff
@@ -164,10 +164,9 @@ export class _DisplayTable extends React.Component<
     if (nextProps !== prevState.initial_props) {
       const new_default_state = get_default_state_from_props(nextProps);
 
-      const prev_sort_still_valid = _.includes(
-        new_default_state.visible_col_keys,
-        prevState.sort_by
-      );
+      const prev_sort_still_valid =
+        !nextProps.unsorted_initial &&
+        _.includes(new_default_state.visible_col_keys, prevState.sort_by);
```

Comment:
The `!nextProps.unsorted_initial` check is to make `sort_by` null when you switch from grouping by one of the dimensions to grouping by `all`. Previously if I switched from grouping by Executive Level to grouping by `all` the Executive Level column would still be sorted. But now that I think about it adding the `!nextProps.unsorted_initial` check always makes`prev_sort_still_valid` false

Reply:
Should changing grouping effect sorting? I think sorting controls and grouping controls being disjoint would be cleaner for the code and have less potentially unexpected behaviour for the users. They might even want to sort by one of the value columns the grouping will be summing up, not the grouped by column itself, so can't say for sure that switching sorting on grouping is actually saving them any clicks, I think.

Reply:
Thing is, if grouping doesn't affect sorting then some of the tables will have weird default orders. In Executive Level the Non-Ex row ends up be between the EX 04 and EX 05 rows. The Voted and Statutory Item rows will be in different orders on different tables. Geographic Region and Transfer Payments by Recipient Region will order the rows differently when grouped by region.

Reply:
okay how about setting `unsorted_initial=false`, then even after grouping, it'll still be sorted by whatever it's grouped by (e.g. grouping by EX level), not sure what you meant in voted and stat and grouping by region but seems fine to me

If a user sorted by a specific column, then how it's sorted is on user

I'm just trying to remember if there was a specific reason why we had it `false`, but seems fine enough

Reply:
This is with`unsorted_initial=true`
II don't know if this would be considered a problem, but Geographic Region and Transfer Payments by Recipient Region list the regions in a different order when they're both grouped by region. I think the Report Builder is just more cohesive when rows are listed in the same order between different tables

Geographic Region:
![Screen Shot 2021-09-25 at 4 14 56 PM](https://user-images.githubusercontent.com/49965519/134784982-6c13d0e5-9cfd-47e0-894c-15d091fb4161.png)

Transfer Payments by Recipient Region:
![Screen Shot 2021-09-25 at 4 15 20 PM](https://user-images.githubusercontent.com/49965519/134784987-15e02370-b6c1-4cf9-b761-6b161d7b5d86.png)